src/Compilers/Server/VBCSCompilerTests/Extensions.cs

```diff
@@ -14,16 +14,16 @@ namespace Microsoft.CodeAnalysis.CompilerServer.UnitTests
 {
     internal static class Extensions
     {
-        public static Task<bool> ToTask(this WaitHandle handle, int? timeoutMilliseconds)
+        public static Task ToTask(this WaitHandle handle, int? timeoutMilliseconds)
         {
             RegisteredWaitHandle registeredHandle = null;
-            var tcs = new TaskCompletionSource<bool>();
+            var tcs = new TaskCompletionSource<object>();
             registeredHandle = ThreadPool.RegisterWaitForSingleObject(
                 handle,
-                (_, timedOut) =>
+                (_, timeout) =>
                 {
-                    tcs.TrySetResult(!timedOut);
-                    if (!timedOut)
+                    tcs.TrySetResult(null);
+                    if (registeredHandle is object)
```

Comment:
>registeredHandle [](start = 24, length = 16)

This `if` statement could be written `registeredHandle?.Unregister(waitObject: null);`